app/api/routers/breeze_buddy/merchant_entities/handlers.py

```
@@ -0,0 +1,327 @@
+
+async def get_merchant_entity_by_merchant_id_handler(
+    merchant_id: str, current_user: UserInfo
+) -> MerchantEntityResponse:
+    """Get merchant entity by merchant_id."""
+    try:
+        merchant = await merchant_queries.get_merchant_entity_by_merchant_id(
+            merchant_id
+        )
+
+        if not merchant:
+            raise HTTPException(status_code=404, detail="Merchant entity not found")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)